### PR TITLE
Setup ssl context in Athenz identity provider

### PR DESCRIPTION
**Pull request by tokle**

### CODE REVIEW

File: container-disc/src/main/java/com/yahoo/container/jdisc/athenz/impl/AthenzIdentityProviderImpl.java [hunk 1]
```diff
@@ -105,6 +119,52 @@ public String getService() {
...
+                        null);
+            return sslContext;
+        } catch (NoSuchAlgorithmException | KeyManagementException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private KeyManager[] createKeyManagersWithServiceCertificate() {
+        try {
+                credentialsRetrievedSignal.await();
```
Comment (bjorncs) [inline container-disc/src/main/java/com/yahoo/container/jdisc/athenz/impl/AthenzIdentityProviderImpl.java]: Wrong indentation

File: container-disc/src/main/java/com/yahoo/container/jdisc/athenz/impl/AthenzIdentityProviderImpl.java [hunk 2]
```diff
@@ -105,6 +119,52 @@ public String getService() {
...
+            keyManagerFactory.init(keyStore, new char[0]);
+            return keyManagerFactory.getKeyManagers();
+        } catch (KeyStoreException | NoSuchAlgorithmException | UnrecoverableKeyException | CertificateException | IOException e) {
+            throw new RuntimeException(e);
+        } catch (InterruptedException e) {
+            throw new AthenzIdentityProviderException("Failed to register instance credentials", lastThrowable.get());
+        }
+    }
+
+    private static TrustManager[] createTrustManagersWithAthenzCa() {
```
Comment (bjorncs) [inline container-disc/src/main/java/com/yahoo/container/jdisc/athenz/impl/AthenzIdentityProviderImpl.java]: We should consider loading the truststore only once in the constructor.

**Review by bjorncs — APPROVED** [review]
👍  We should consider rewriting the initial registration to be blocking in the constructor.

File: container-disc/src/main/java/com/yahoo/container/jdisc/athenz/impl/AthenzIdentityProviderImpl.java (hunk 2 above)
Comment (tokle) [reply container-disc/src/main/java/com/yahoo/container/jdisc/athenz/impl/AthenzIdentityProviderImpl.java]: Let's do when we consolidate Athenz trust-/keystore implementations

**Pull request merged**